pkg/scheduler/scheduler.go

```diff
@@ -162,9 +163,11 @@ func WithParallelism(threads int32) Option {
 }
 
 // WithPercentageOfNodesToScore sets percentageOfNodesToScore for Scheduler, the default value is 50
-func WithPercentageOfNodesToScore(percentageOfNodesToScore int32) Option {
+func WithPercentageOfNodesToScore(percentageOfNodesToScore *int32) Option {
```

Comment:
Let's add a test for this function, to check that if the value is nil, the option is 0.

Reply:
Added a test function for it.